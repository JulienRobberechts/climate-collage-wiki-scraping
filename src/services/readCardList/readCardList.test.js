/**
 * @jest-environment node
 */
const { readCardList } = require('./readCardList');

describe('readCardList', () => {
  it('read card list FR', async () => {
    const lang = 'fr';
    const cards = await readCardList(lang);
    expect(cards).toMatchSnapshot();
  });
  it('read card list EN', async () => {
    const lang = 'en';
    const cards = await readCardList(lang);
    expect(cards).toMatchSnapshot();
  });
});
