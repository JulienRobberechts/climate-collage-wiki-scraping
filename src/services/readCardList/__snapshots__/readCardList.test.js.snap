// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readCardList read card list 1`] = `
Array [
  Object {
    "cardNum": 1,
    "cardSet": 1,
    "cardWikiInternalName": "Fr-fr adulte carte 1 activités humaines",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_1_activit%C3%A9s_humaines",
    "title": "Activités humaines",
  },
  Object {
    "cardNum": 2,
    "cardSet": 2,
    "cardWikiInternalName": "Fr-fr adulte carte 2 industrie",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_2_industrie",
    "title": "Industrie",
  },
  Object {
    "cardNum": 3,
    "cardSet": 2,
    "cardWikiInternalName": "Fr-fr adulte carte 3 utilisation des bâtiments",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_3_utilisation_des_b%C3%A2timents",
    "title": "Utilisation des bâtiments",
  },
  Object {
    "cardNum": 4,
    "cardSet": 2,
    "cardWikiInternalName": "Fr-fr adulte carte 4 transport",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_4_transport",
    "title": "Transport",
  },
  Object {
    "cardNum": 5,
    "cardSet": 1,
    "cardWikiInternalName": "Fr-fr adulte carte 5 énergies fossiles",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_5_%C3%A9nergies_fossiles",
    "title": "Énergies fossiles",
  },
  Object {
    "cardNum": 6,
    "cardSet": 2,
    "cardWikiInternalName": "Fr-fr adulte carte 6 déforestation",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_6_d%C3%A9forestation",
    "title": "Déforestation",
  },
  Object {
    "cardNum": 7,
    "cardSet": 1,
    "cardWikiInternalName": "Fr-fr adulte carte 7 émissions de co2",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_7_%C3%A9missions_de_co2",
    "title": "Émissions de CO2",
  },
  Object {
    "cardNum": 8,
    "cardSet": 2,
    "cardWikiInternalName": "Fr-fr adulte carte 8 agriculture",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_8_agriculture",
    "title": "Agriculture",
  },
  Object {
    "cardNum": 9,
    "cardSet": 2,
    "cardWikiInternalName": "Fr-fr adulte carte 9 autres ges",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_9_autres_ges",
    "title": "Autres GES",
  },
  Object {
    "cardNum": 10,
    "cardSet": 3,
    "cardWikiInternalName": "Fr-fr adulte carte 10 aérosols",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_10_a%C3%A9rosols",
    "title": "Aérosols",
  },
  Object {
    "cardNum": 11,
    "cardSet": 2,
    "cardWikiInternalName": "Fr-fr adulte carte 11 concentration en co2",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_11_concentration_en_co2",
    "title": "Concentration en CO2 (ppm)",
  },
  Object {
    "cardNum": 12,
    "cardSet": 2,
    "cardWikiInternalName": "Fr-fr adulte carte 12 puits de carbone",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_12_puits_de_carbone",
    "title": "Puits de carbone",
  },
  Object {
    "cardNum": 13,
    "cardSet": 1,
    "cardWikiInternalName": "Fr-fr adulte carte 13 effet de serre additionnel",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_13_effet_de_serre_additionnel",
    "title": "Effet de serre additionnel",
  },
  Object {
    "cardNum": 14,
    "cardSet": 3,
    "cardWikiInternalName": "Fr-fr adulte carte 14 bilan énergétique",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_14_bilan_%C3%A9nerg%C3%A9tique",
    "title": "Bilan énergétique",
  },
  Object {
    "cardNum": 15,
    "cardSet": 3,
    "cardWikiInternalName": "Fr-fr adulte carte 15 forçage radiatif",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_15_for%C3%A7age_radiatif",
    "title": "Forçage radiatif",
  },
  Object {
    "cardNum": 16,
    "cardSet": 3,
    "cardWikiInternalName": "Fr-fr adulte carte 16 fonte des glaciers",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_16_fonte_des_glaciers",
    "title": "Fonte des glaciers",
  },
  Object {
    "cardNum": 17,
    "cardSet": 3,
    "cardWikiInternalName": "Fr-fr adulte carte 17 hausse température eau",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_17_hausse_temp%C3%A9rature_eau",
    "title": "Hausse de la température de l'eau",
  },
  Object {
    "cardNum": 18,
    "cardSet": 1,
    "cardWikiInternalName": "Fr-fr adulte carte 18 fonte de la banquise",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_18_fonte_de_la_banquise",
    "title": "Fonte de la banquise",
  },
  Object {
    "cardNum": 19,
    "cardSet": 3,
    "cardWikiInternalName": "Fr-fr adulte carte 19 fonte des calottes glaciaires",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_19_fonte_des_calottes_glaciaires",
    "title": "Fonte des calottes glaciaires",
  },
  Object {
    "cardNum": 20,
    "cardSet": 3,
    "cardWikiInternalName": "Fr-fr adulte carte 20 perturbation cycle eau",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_20_perturbation_cycle_eau",
    "title": "Perturbation du cycle de l'eau",
  },
  Object {
    "cardNum": 21,
    "cardSet": 1,
    "cardWikiInternalName": "Fr-fr adulte carte 21 hausse température",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_21_hausse_temp%C3%A9rature",
    "title": "Hausse de la température",
  },
  Object {
    "cardNum": 22,
    "cardSet": 1,
    "cardWikiInternalName": "Fr-fr adulte carte 22 montée des eaux",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_22_mont%C3%A9e_des_eaux",
    "title": "Montée des eaux",
  },
  Object {
    "cardNum": 23,
    "cardSet": 4,
    "cardWikiInternalName": "Fr-fr adulte carte 23 problèmes de calcification",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_23_probl%C3%A8mes_de_calcification",
    "title": "Problèmes de calcification",
  },
  Object {
    "cardNum": 24,
    "cardSet": 2,
    "cardWikiInternalName": "Fr-fr adulte carte 24 acidification océan",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_24_acidification_oc%C3%A9an",
    "title": "Acidification de l'océan",
  },
  Object {
    "cardNum": 25,
    "cardSet": 4,
    "cardWikiInternalName": "Fr-fr adulte carte 25 biodiversité terrestre",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_25_biodiversit%C3%A9_terrestre",
    "title": "Biodiversité terrestre",
  },
  Object {
    "cardNum": 26,
    "cardSet": 4,
    "cardWikiInternalName": "Fr-fr adulte carte 26 crues",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_26_crues",
    "title": "Crues",
  },
  Object {
    "cardNum": 27,
    "cardSet": 4,
    "cardWikiInternalName": "Fr-fr adulte carte 27 biodiversité marine",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_27_biodiversit%C3%A9_marine",
    "title": "Biodiversité marine",
  },
  Object {
    "cardNum": 28,
    "cardSet": 5,
    "cardWikiInternalName": "Fr-fr adulte carte 28 vecteurs de maladie",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_28_vecteurs_de_maladie",
    "title": "Vecteurs de maladie",
  },
  Object {
    "cardNum": 29,
    "cardSet": 4,
    "cardWikiInternalName": "Fr-fr adulte carte 29 ptéropodes et coccolithophores",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_29_pt%C3%A9ropodes_et_coccolithophores",
    "title": "Ptéropodes et coccolithophores",
  },
  Object {
    "cardNum": 30,
    "cardSet": 4,
    "cardWikiInternalName": "Fr-fr adulte carte 30 sécheresses",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_30_s%C3%A9cheresses",
    "title": "Sécheresses",
  },
  Object {
    "cardNum": 31,
    "cardSet": 5,
    "cardWikiInternalName": "Fr-fr adulte carte 31 ressources eau douce",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_31_ressources_eau_douce",
    "title": "Ressources en eau douce",
  },
  Object {
    "cardNum": 32,
    "cardSet": 5,
    "cardWikiInternalName": "Fr-fr adulte carte 32 baisse rendements agricoles",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_32_baisse_rendements_agricoles",
    "title": "Baisse des rendements agricoles",
  },
  Object {
    "cardNum": 33,
    "cardSet": 4,
    "cardWikiInternalName": "Fr-fr adulte carte 33 submersions",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_33_submersions",
    "title": "Submersions",
  },
  Object {
    "cardNum": 34,
    "cardSet": 4,
    "cardWikiInternalName": "Fr-fr adulte carte 34 cyclones",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_34_cyclones",
    "title": "Cyclones",
  },
  Object {
    "cardNum": 35,
    "cardSet": 5,
    "cardWikiInternalName": "Fr-fr adulte carte 35 incendies",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_35_incendies",
    "title": "Incendies",
  },
  Object {
    "cardNum": 36,
    "cardSet": 5,
    "cardWikiInternalName": "Fr-fr adulte carte 36 canicules",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_36_canicules",
    "title": "Canicules",
  },
  Object {
    "cardNum": 37,
    "cardSet": 5,
    "cardWikiInternalName": "Fr-fr adulte carte 37 famine",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_37_famine",
    "title": "Famine",
  },
  Object {
    "cardNum": 38,
    "cardSet": 5,
    "cardWikiInternalName": "Fr-fr adulte carte 38 santé humaine",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_38_sant%C3%A9_humaine",
    "title": "Santé humaine",
  },
  Object {
    "cardNum": 39,
    "cardSet": 5,
    "cardWikiInternalName": "Fr-fr adulte carte 39 réfugiés climatiques",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_39_r%C3%A9fugi%C3%A9s_climatiques",
    "title": "Réfugiés climatiques",
  },
  Object {
    "cardNum": 40,
    "cardSet": 5,
    "cardWikiInternalName": "Fr-fr adulte carte 40 conflits armés",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_40_conflits_arm%C3%A9s",
    "title": "Conflits armés",
  },
  Object {
    "cardNum": 41,
    "cardSet": 5,
    "cardWikiInternalName": "Fr-fr adulte carte 41 permafrost",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_41_permafrost",
    "title": "Permafrost",
  },
  Object {
    "cardNum": 42,
    "cardSet": 5,
    "cardWikiInternalName": "Fr-fr adulte carte 42 hydrates de méthane",
    "cardWikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_42_hydrates_de_m%C3%A9thane",
    "title": "Hydrates de méthane",
  },
]
`;
